tests/unit/acquisition/test_function.py

```diff
@@ -546,20 +584,76 @@ def test_batch_reparametrization_sampler_samples_are_distinct_for_new_instances(
     sampler = BatchReparametrizationSampler(100, QuadraticMeanAndRBFKernel())
 
-    with pytest.raises((ValueError, tf.errors.InvalidArgumentError)):
+    with pytest.raises(TF_DEBUGGING_ERROR_TYPES):
         sampler.sample(tf.constant([[0.0]]), jitter=-1e-6)
 
 
 def test_batch_reparametrization_sampler_sample_raises_for_inconsistent_batch_size() -> None:
     sampler = BatchReparametrizationSampler(100, QuadraticMeanAndRBFKernel())
     sampler.sample(tf.constant([[0.0], [1.0], [2.0]]))
 
-    with pytest.raises((ValueError, tf.errors.InvalidArgumentError)):
+    with pytest.raises(TF_DEBUGGING_ERROR_TYPES):
         sampler.sample(tf.constant([[0.0], [1.0]]))
+
+
+@pytest.mark.parametrize("sample_size", [-2, 0])
+def test_batch_monte_carlo_expected_improvement_raises_for_invalid_sample_size(
+    sample_size: int,
+) -> None:
+    with pytest.raises(TF_DEBUGGING_ERROR_TYPES):
+        BatchMonteCarloExpectedImprovement(sample_size)
+
+
+def test_batch_monte_carlo_expected_improvement_raises_for_invalid_jitter() -> None:
+    with pytest.raises(TF_DEBUGGING_ERROR_TYPES):
+        BatchMonteCarloExpectedImprovement(100, jitter=-1.0)
+
+
+def test_batch_monte_carlo_expected_improvement_raises_for_empty_data() -> None:
+    builder = BatchMonteCarloExpectedImprovement(100)
+    data = Dataset(tf.zeros([0, 2]), tf.zeros([0, 1]))
+    model = QuadraticMeanAndRBFKernel()
+    with pytest.raises(TF_DEBUGGING_ERROR_TYPES):
+        builder.prepare_acquisition_function(data, model)
+
+
+def test_batch_monte_carlo_expected_improvement_raises_for_model_with_wrong_event_shape() -> None:
+    builder = BatchMonteCarloExpectedImprovement(100)
+    data = mk_dataset([[0.0, 0.0]], [[0.0, 0.0]])
+    model = _dim_two_gp()
+    with pytest.raises(TF_DEBUGGING_ERROR_TYPES):
+        builder.prepare_acquisition_function(data, model)
+
+
+@random_seed
+def test_batch_monte_carlo_expected_improvement_can_reproduce_ei() -> None:
+    known_query_points = tf.random.uniform([5, 2], dtype=tf.float64)
+    data = Dataset(known_query_points, quadratic(known_query_points))
+    model = QuadraticMeanAndRBFKernel()
+    batch_ei = BatchMonteCarloExpectedImprovement(10_000).prepare_acquisition_function(data, model)
+    ei = ExpectedImprovement().prepare_acquisition_function(data, model)
+    xs = tf.random.uniform([3, 5, 1, 2], dtype=tf.float64)
+    npt.assert_allclose(batch_ei(xs), ei(tf.squeeze(xs, -2)), rtol=0.03)  # todo a little high
+
+
+@random_seed
+def test_batch_monte_carlo_expected_improvement() -> None:
+    xs = tf.random.uniform([3, 5, 7, 2], dtype=tf.float64)
+    model = QuadraticMeanAndRBFKernel()
+
+    mean, cov = model.predict_joint(xs)
+    mvn = tfp.distributions.MultivariateNormalFullCovariance(tf.linalg.matrix_transpose(mean), cov)
+    mvn_samples = mvn.sample(10_000)
+    expected = tf.reduce_mean(tf.reduce_max(tf.maximum(0.09 - mvn_samples, 0.0), axis=-1), axis=0)
```

Comment:
are we sure that eta=0.09 for acq?

Reply:
I would rather use `eta = tf.reduce_min(mean, axis=0)` instead of the hard-coded 0.09 above

Reply:
`QuadraticMeanAndRBFKernel` is a `GaussianProcess`, which outputs precisely the mean you tell it to. In this case, that's the quadratic function, so 0.3 ** 2

Reply:
I could make that clearer by changing the 0.09 in the data below to have some noise on it, to make it clear they're not the same thing?

Reply:
Don't add noise! this is fine. I just find it hard to follow, there's no visible connexion between the model mean and 0.09. 
I may just not be used to the way tests ought to be written

Reply:
i could
```
min_model_mean = 0.09
... (min_model_mean - mvn_samples) ...
```

Reply:
I like that